airflow/ui/src/components/TogglePause.tsx

```diff
@@ -42,12 +43,19 @@ export const TogglePause = ({ dagDisplayName, dagId, isPaused, skipConfirm }: Pr
   const { onClose, onOpen, open } = useDisclosure();
 
   const onSuccess = async () => {
-    await queryClient.invalidateQueries({
-      queryKey: [useDagServiceGetDagsKey],
-    });
+    // We can just update dag details
+    queryClient.setQueryData<DAGDetailsResponse>(UseDagServiceGetDagDetailsKeyFn({ dagId }), (oldData) =>
+      oldData
+        ? {
+            ...oldData,
+            is_paused: !oldData.is_paused,
+          }
+        : undefined,
+    );
```

Comment:
I'm not super fan of this.

If we add some business logic and side effect on pausing / unpausing a dag. That will start failing.

We are unnecessary coupling the back and the front on that part. I think an extra GET is really not that expensive and worth the trouble. (if we generalize that pattern we might end up with issues)

Reply:
Ok I'll drop it